src/nodes/typeGuards/compound.ts

```diff
@@ -61,6 +61,23 @@ export function isJsxTagNamePropertyAccess(
 	);
 }
 
+/**
+ * ts.NamedDeclaration but the name property isn't optional.
+ */
+export interface NamedDeclaration extends ts.NamedDeclaration {
```

Comment:
In order to avoid confusion, we might want to give this type a different name such as `NamedDeclarationWithName`